src/ring_theory/fractional_ideal.lean

```diff
@@ -183,10 +184,12 @@ also called `coe_to_submodule` in theorem names.
 
 This map is available as a ring hom, called `fractional_ideal.coe_ideal_hom`.
 -/
+def coe_ideal : ideal R → fractional_ideal S P := λ I,
+⟨coe_submodule P I,
+  is_fractional_of_le_one _ $ by simpa using coe_submodule_mono P (le_top : I ≤ ⊤)⟩
```

Comment:
There were quite a few instances previously where I want to use this as a function e.g. in `congr_arg coe_ideal`, and writing `coe : ideal R → fractional_ideal S P` everywhere is very wordy, especially since `R`, `S`, and `P` are long in my case.

Reply:
You mean you needed it previously but no longer need it? Then I suggest just removing it (inlining it like before); indeed I didn't find any usage in your open PRs. In mathlib we usually use coercion as the simp normal form, and replacing it by `coe_ideal` changes the syntactic form of the original lemmas and breaks the norm.
I mention Lean 4 because (IIUC) coercions will be directly elaborated into the underlying function, so a lemma stated using coercion arrow will be syntactically the same as one stated using the function (e.g. `coe_ideal`); in fact there's no longer the function `coe` that takes an instance argument, so you can no longer write those `injective coe` lemmas.

I'd be surprised if `congr_arg coe_ideal` doesn't require type ascription but `congr_arg coe` does; I guess you could also use `congr_arg coe_ideal_hom`? And I don't quite see from this use case why you want the two `coe_ideal_injective` lemmas to be stated using `coe_ideal`.